### PR TITLE
Render: Support 3d hexagons

### PR DESCRIPTION
**Pull request by JettChenT**
#869 

This adds support for 3d hexagons. For example, the following code:
```
hexagon: {shape: "hexagon"}
hexagon.style.3d: true
```
will produce:
CleanShot 2023-02-25 at 11 33 15@2x

This example can be ran via `./ci/e2ereport.sh -test-case=hexagon_3d`

### CODE REVIEW

**Comment by alixander** [comment]
@katwangy can you check if the dimensions of the 3d stuff look right to u?

**Comment by alixander** [comment]
@JettChenT could you sign your commits please? https://github.com/terrastruct/d2/blob/master/docs/CONTRIBUTING.md#logistics

**Review by alixander — CHANGES_REQUESTED** [review]
ty @JettChenT ! code looks good to me, and quite reusable for future 3d implementations.

i pulled and ran `go install .`, and played around with it.

the default themed color looks a bit different (this one in sketched mode, but i don't think it made a difference)

Screen Shot 2023-02-25 at 8 04 01 AM

```
hexagon: {
  shape: hexagon
  style.3d: true
}


rect: {
  shape: rectangle
  style.3d: true
}

hexagon -> rect
```

and then when i add a fill, it seems to be ignored (can you add an e2e test that has all 3d shapes fill and stroke set?):

Screen Shot 2023-02-25 at 8 05 03 AM

```
hexagon: {
  shape: hexagon
  style.3d: true
  style.fill: honeydew
}


rect: {
  shape: rectangle
  style.3d: true
  style.fill: honeydew
}

hexagon -> rect
```

**Comment by JettChenT** [comment]
Hi! I've set up commit signing, and fill should now work as expected(e2e test accesible via `3d_fill_and_stroke`):
sketch exp

However, the default 3d appearance of 3d Hexagon is still different in color from the 3d rectangle.
sketch exp
CleanShot 2023-02-26 at 10 49 45@2x

This is most likely due to the default fill color for them being different:
CleanShot 2023-02-26 at 10 52 36@2x

CleanShot 2023-02-26 at 10 51 41@2x

File: d2renderers/d2svg/d2svg.go [hunk 1]
```diff
@@ -746,6 +746,128 @@ func render3dRect(targetShape d2target.Shape) string {
...
+	}
+	halfYFactor := 43.6 / 87.3
+
+	// draw border all in one path to prevent overlapping sections
+	var borderSegments []string
+	borderSegments = append(borderSegments,
+		moveTo(d2target.Point{X: scale(targetShape.Width, 0.25), Y: 0}),
+	)
+	Y_OFFSET := d2target.THREE_DEE_OFFSET / 2
+	for _, v := range []d2target.Point{
```
Comment (alixander) [inline d2renderers/d2svg/d2svg.go:767]: maybe leave a comment on which part of the 3d this is drawing. e.g. the top side vs the right side

File: d2renderers/d2svg/d2svg.go [hunk 2]
```diff
@@ -746,6 +746,128 @@ func render3dRect(targetShape d2target.Shape) string {
...
+		{X: scale(targetShape.Width, 0.75), Y: targetShape.Height},
+		{X: targetShape.Width, Y: scale(targetShape.Height, halfYFactor)},
+		{X: scale(targetShape.Width, 0.75), Y: 0},
+		{X: scale(targetShape.Width, 0.25), Y: 0},
+	} {
+		sidePoints = append(sidePoints,
+			fmt.Sprintf("%d,%d", v.X+targetShape.Pos.X, v.Y+targetShape.Pos.Y),
+		)
+	}
+	// TODO make darker color part of the theme? or just keep this bypass
```
Comment (alixander) [inline d2renderers/d2svg/d2svg.go:856]: bit of an ongoing debate hah, i think this is fine for now. https://github.com/terrastruct/d2/issues/896

File: d2renderers/d2svg/d2svg.go [hunk 3]
```diff
@@ -958,6 +1080,39 @@ func drawShape(writer io.Writer, targetShape d2target.Shape, sketchRunner *d2ske
 			}
 		}
 	case d2target.ShapeText, d2target.ShapeCode:
+	case d2target.ShapeHexagon:
```
Comment (alixander) [inline d2renderers/d2svg/d2svg.go:1083]: can u put this above the line above? i think slightly nicer to have the "do nothing" cases next to default and not sandwiched between the "do something" cases

**Review by alixander — CHANGES_REQUESTED** [review]
excellent! looks great.

Screen Shot 2023-02-25 at 8 06 42 PM

i did notice that the more square hexagon looks better than the wide ones. separate issue from this though, so made another to track: https://github.com/terrastruct/d2/issues/904

Could you squash your commits or re-commit to sign those previous ones please? then i think it's good to go! thank you!

**Comment by katwangy** [comment]
Possible to make it thicker so the depth is more on par with the cube? The top part of the hexagon is looking a little thin

File: d2renderers/d2svg/d2svg.go (hunk 1 above)
Comment (JettChenT) [reply d2renderers/d2svg/d2svg.go:768]: got it. I've added the comments

File: d2renderers/d2svg/d2svg.go (hunk 3 above)
Comment (JettChenT) [reply d2renderers/d2svg/d2svg.go:1083]: Alright! I've changed the order

File: d2renderers/d2svg/d2svg.go (hunk 2 above)
Comment (JettChenT) [reply d2renderers/d2svg/d2svg.go:857]: got it.

**Comment by JettChenT** [comment]
Hi, I've squashed the commits and made the corresponding changes. Not quite sure why the CI fails on `COLOR=1 ./make.sh` though, as it works on my local machine.

@katwangy would the following be better?(adjusting Y offset from 1/2 of 3D offset to 2/3 of 3D offset)
CleanShot 2023-02-27 at 09 58 48@2x

**Review by alixander — APPROVED** [review]
perfect, thanks @JettChenT !!

**Pull request merged**